src/leo-server-app/src/routes/schedule.ts

```diff
@@ -80,6 +96,76 @@ router.post("/createSchedule", async (req: CreateScheduleProp, res: any) => {
   res.status(201).json(resObj);
 });
 
+const isAdminCheck = async (userId: string) => {
+  const userRecord = await User.findById(userId);
+  console.log(userRecord?.role);
+  return userRecord?.role === UserRole.ADMIN;
+};
+
+const checkSatellitePermissionList = async (
+  satelliteId: string,
+  command: string
+) => {
+  // Fetch satellite
+  const satellite = await Satellite.findById(satelliteId);
+
+  const isValid = satellite?.validCommands.includes(command);
+  return isValid;
+};
+
+router.patch(
+  "/updateScheduledCommand",
+  async (req: UpdateScheduleProp, res: any) => {
+    const { userId, commandId, satelliteId, command } = req.query;
+
+    // Validation
+    if (
+      !mongoose.isValidObjectId(userId) ||
+      !mongoose.isValidObjectId(commandId)
+    ) {
+      return res.status(500).json({ error: "Invalid IDs" });
+    }
+
+    const userRecord = await User.findById(userId);
+
+    if (!userRecord) {
+      return res.status(500).json({ error: "User does not exist" });
+    }
+
+    // Check if user has permission
+    if (userRecord.role !== UserRole.ADMIN) {
+      const commandRecord = await Command.findById(commandId);
+      if (commandRecord?.userId?.toString() !== userId) {
+        return res.status(500).json({ error: "Invalid Credentials" });
+      }
+    }
+
+    // Add validation for invalid command sequence based on satellite and user permissions
+    // Test 1 - Check if command exists in the satellite's list of command sequences
+    const isCommandInSatelliteCriteria = await checkSatellitePermissionList(
+      satelliteId,
+      command
+    );
+
+    if (!isCommandInSatelliteCriteria) {
+      return res.status(500).json({ error: "Invalid command sequence" });
+    }
+
+    // Test 2 - Check if command exists in the user's permission list for satellite unless they are admin
+
+    // Update command record
+    const updatedCommand = await Command.findByIdAndUpdate(
+      commandId,
+      {
+        command,
+      },
+      { new: true }
+    ).exec();
+
+    return res.json({ message: "Updated command", updatedCommand });
+  }
+);
+
```

Comment:
tests should be a ina separate testing file, not sure what its doing here

Reply:
oh mb these are validation checks, just labelled them as tests, lemme add betteer comment